ast/json_test.go

```diff
@@ -157,6 +157,308 @@ func TestJSONMarshal(t *testing.T) {
 			},
 			want: `{"type":"BuiltinStatement","id":{"type":"Identifier","name":"task"}}`,
 		},
+		{
+			name: "NamedType",
+			node: &ast.NamedType{
+				BaseNode: ast.BaseNode{},
+				ID: &ast.Identifier{
+					BaseNode: ast.BaseNode{},
+					Name:     "int",
+				},
+			},
+			want: `{"type":"NamedType","name":{"type":"Identifier","name":"int"}}`,
+		},
+		{
+			name: "TvarType",
+			node: &ast.TvarType{
+				BaseNode: ast.BaseNode{},
+				ID: &ast.Identifier{
+					BaseNode: ast.BaseNode{},
+					Name:     "A",
+				},
+			},
+			want: `{"type":"TvarType","name":{"type":"Identifier","name":"A"}}`,
+		},
+		{
+			name: "ArrayType",
+			node: &ast.ArrayType{
+				BaseNode: ast.BaseNode{},
+				ElementType: &ast.ArrayType{
+					BaseNode: ast.BaseNode{},
+					ElementType: &ast.NamedType{
+						BaseNode: ast.BaseNode{},
+						ID: &ast.Identifier{
+							BaseNode: ast.BaseNode{},
+							Name:     "int",
+						},
+					},
+				},
+			},
+			// r#"{"type":"ArrayType","element":{"type":"ArrayType","element":{"type":"NamedType","name":{"type":"Identifier","name":"A"}}}}"#
+			want: `{"type":"ArrayType","elementType":{"type":"ArrayType","elementType":{"type":"NamedType","name":{"type":"Identifier","name":"int"}}}}`,
```

Comment:
It looks like the json tag needs to be `element` here. The rust json comment is very helpful by the way.